xbmc/Application.cpp

```diff
@@ -4211,9 +4213,9 @@ bool CApplication::OnMessage(CGUIMessage& message)
         CGUIMessage msg(GUI_MSG_PLAYLISTPLAYER_CHANGED, 0, 0, g_playlistPlayer.GetCurrentPlaylist(), param, item);
         g_windowManager.SendThreadMessage(msg);
         g_playlistPlayer.SetCurrentSong(m_nextPlaylistItem);
-        *m_itemCurrentFile = *item;
+        m_itemCurrentFile = item;
```

Comment:
are you sure that this doesn't require a deep copy as you've done above (and as was originally achieved through the dereferencing of *m_itemCurrentFile...)